kv/kv.go

```diff
@@ -13,25 +13,68 @@
 
 package kv
 
+import "bytes"
+
+// EncodedKey represents encoded key in low-level storage engine.
+type EncodedKey []byte
+
+// Key represents high-level Key type.
+type Key []byte
+
+// Next returns the next key in byte-order.
+func (k Key) Next() Key {
+	// add \x0 to the end of key
+	buf := make([]byte, len([]byte(k))+1)
+	copy(buf, []byte(k))
+	return buf
+}
+
+// Cmp returns the comparison result of two key, if A > B returns 1, A = B
+// returns 0, if A < B returns -1.
```

Comment:
The result will be 0 if a==b, -1 if a < b, and +1 if a > b. 
See http://golang.org/pkg/bytes/#Compare